### ISSUE TITLE
halide/gcc_linker_stderr PR causes broken builds on macOS

### ISSUE THREAD

**Issue by shoaibkamil** [halide/Halide#2940]
#2884 causes broken builds on macOS 10.13, XCode 9.3:

```
IRPrinter test passed
CodeGen_C test passed
ir_equality_test passed
Bounds test passed
expr_match test passed
deinterleave_vector test passed
modulus_remainder test passed
common_subexpression_elimination test passed
Solve test passed
Target test passed
is_monotonic test passed
Split predicate test passed
Interval test passed
Associativity test passed
Generator test passed

bash: line 1011: 1011n: command not found
bash: line 1012: 1012n: command not found
bash: line 1013: 1013n: command not found
bash: line 1014: 1014n: command not found
ar: /libLLVMX86Utils.a: No such file or directory
mv: rename X86ShuffleDecode.cpp.o to llvm_1015_X86ShuffleDecode.cpp.o: No such file or directory
# Archive together all the halide and llvm object files
```

The build completes, but the resulting libHalide is broken and will give missing symbol errors when linking anything requiring libHalide.

Sigh.  This is because some LLVM installs need libcurses, and so the `ar` output has:
```
Undefined symbols for architecture x86_64:
  "_del_curterm", referenced from:
      llvm::sys::Process::FileDescriptorHasColors(int) in libLLVMSupport.a(Process.cpp.o)
  "_set_curterm", referenced from:
      llvm::sys::Process::FileDescriptorHasColors(int) in libLLVMSupport.a(Process.cpp.o)
  "_setupterm", referenced from:
      llvm::sys::Process::FileDescriptorHasColors(int) in libLLVMSupport.a(Process.cpp.o)
  "_tigetnum", referenced from:
      llvm::sys::Process::FileDescriptorHasColors(int) in libLLVMSupport.a(Process.cpp.o)
ld: symbol(s) not found for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
```
and the lines with `libLLVM` now appear in the LLVM object list.

**Comment by shoaibkamil** [comment]
Upon further investigation, the rule that creates the object list uses `LLVM_STATIC_LIBS` which is `llvm-config --link-static`, but the required system libraries come from `llvm-config --system-libs`.

**Comment by shoaibkamil** [comment]
Fixed in #2943.

**Comment by luciansmith** [comment]
Just a note to say thanks--I ran into this exact issue on my own llvm-using project, and you provided the exact information I needed to solve it there, too.
